BootloaderCommonPkg/Include/Library/SpiBlockIoLib.h

```diff
@@ -15,7 +15,7 @@
 #include <BlockDevice.h>
 #include <Service/SpiFlashService.h>
 
-#define SPI_BLOCK_SIZE 0x100
+#define SPI_BLOCK_SIZE 0x1

```

Comment:
Any specific reason we want to use 1 as block size ?    Any problem with 0x100 ?   The intension is to ensuare the components are aligned on flash properly.

Reply:
SpiReadBlocks function is calling SPI Read API as below 
mSpiService->SpiRead (RegionType, (UINT32) (Address * SPI_BLOCK_SIZE), (UINT32)BufferSize, Buffer);
Address is always multiplied with Block Size which is not correct in case of LBA address is passed. 
i will revert to SPI block size to 0x100 and modify SpiReadBlocks function as below. 
mSpiService->SpiRead (RegionType, (UINT32) (Address ), (UINT32)BufferSize, Buffer); is it ok ?

Reply:
Yes, that is fine.   The address is actually  LBA address (Block Index) for SPI flash.